app/src/main/java/com/android/voyageur/ui/overview/Overview.kt

```diff
@@ -91,21 +96,42 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
  * @param navigationActions Actions to handle navigation between screens.
  * @param userViewModel The ViewModel containing the state and logic for user data.
  */
-@OptIn(ExperimentalCoroutinesApi::class, ExperimentalMaterial3Api::class)
+@OptIn(ExperimentalCoroutinesApi::class)
 @Composable
 fun OverviewScreen(
     tripsViewModel: TripsViewModel,
     navigationActions: NavigationActions,
     userViewModel: UserViewModel,
 ) {
-  val trips by tripsViewModel.trips.collectAsState()
+  val unfilteredTrips by tripsViewModel.trips.collectAsState()
   val isLoadingUser by userViewModel.isLoading.collectAsState()
   val isLoadingTrip by tripsViewModel.isLoading.collectAsState()
   var isLoading = false
   val status by connectivityState()
   val context = LocalContext.current
   val isConnected = status === ConnectionState.Available
   var searchQuery by remember { mutableStateOf("") }
+  var showOnlyFavorites by remember { mutableStateOf(false) }
+  val user by userViewModel.user.collectAsState()
+  if (user == null) {
```

Comment:
I am wondering if here this is the best case of handling `user == null`. I assume we re-load the Overview screen to wait for the user data to be fetched, but can it be possible that the user data is not fetched and the Overview Screen keeps re-loading? LMK your thoughts on this, maybe I got this wrong.

Reply:
Also, to check if the user is logged in you should use the `Firebase.auth.uid` string

Reply:
I agree this wasn't the best approach for a null user handling. After discussing this with Mihai offline, I changed the behavior to just return (not display anything). When the user is loaded and is not null anymore, the screen will recompose.